runner/metrics/metrics.go

```diff
@@ -0,0 +1,70 @@
+package metrics
+
+import (
+	"log/slog"
+	"time"
+
+	"github.com/cloudbase/garm/auth"
+	"github.com/cloudbase/garm/runner"
+)
+
+func CollectObjectMetric(r *runner.Runner, timer *time.Ticker) {
+
+	ctx := auth.GetAdminContext()
+
+	controllerInfo, err := r.GetControllerInfo(ctx)
+	if err != nil {
+		slog.With(slog.Any("error", err)).ErrorContext(ctx, "cannot get controller info")
+	}
+
+	go func() {
+		// we wan't to initiate the collection immediately
+		for ; true; <-timer.C {
```

Comment:
This should probably have a `select{}` block and return if `ctx.Done()` returns.

Reply:
done with https://github.com/cloudbase/garm/pull/216/commits/97f172eb513bda47d6658b81fd9bf262d687aa98